### PR TITLE
use immutable pointers instead of weakref and wrap in testset

### PR DESCRIPTION
**Pull request by vchuravy**
Should address #66. I am currently running the tests in a loop to see if the flaky tests are fixed.

### CODE REVIEW

**Comment by vchuravy** [comment]
Well I don;t see the error locally with 0.5 grrr....

File: src/pointer.jl [hunk 1]
```diff
@@ -30,23 +30,25 @@ rawpointer(p::CudaPtr) = p
...
-    cuda_ptrs[WeakRef(cptr)] = device()
     cptr
 end
 malloc(nbytes::Integer) = malloc(UInt8, nbytes)
 
 function free{T}(p::CudaPtr{T})
     cnull = unsafe_convert(Ptr{T}, C_NULL)
-    if p.ptr != cnull && haskey(cuda_ptrs, p)
-        delete!(cuda_ptrs, p)
+    if p.ptr != cnull && haskey(cuda_ptrs, p.ptr)
```
Comment (musm) [inline src/pointer.jl]: C_NULL ?

**Comment by musm** [comment]
Unfortunately this also doesn't seem to work on 0.6, e.g. running the troublesome test multiple times eventually triggers the problem.

**Comment by vchuravy** [comment]
Yeah I have to check again if I can reproduce the error locally. I could not last time.

@musm If you have some time to investigate and maybe come up with a fix that would really be great.

**Pull request closed without merging**